### PR TITLE
SVCB: Correctly parse and print unknown params

### LINKED ISSUE #9829: sdig: Unrecognized HTTPS/SVCB param value can print non-printable characters
- Program: Authoritative
 - Issue type: Bug report

### Short description
If you `sdig` an `HTTPS` record (or probably an `SVCB` record) with an unrecognized parameter, the param value is displayed without being escaped, and can therefore write non-printable characters to your terminal.

It's a (variant of a) character string, so I think it should be `\123` escaped.

### Environment
 - Operating system: Ubuntu 16.04 >_>
 - Software version: 4.5.0+master.0.g2717eea26-1pdns.xenial
 - Software source: PowerDNS repository

### Steps to reproduce
1. `sdig ::1 53 odoh.cloudflare-dns.com HTTPS recurse | tail -2 | hd`

### Expected behaviour
Some kind of text output.

### Actual behaviour
```
$ sdig ::1 53 odoh.cloudflare-dns.com HTTPS recurse
Reply to question for qname='odoh.cloudflare-dns.com.', qtype=HTTPS
Rcode: 0 (No Error), RD: 1, QR: 1, TC: 0, AA: 0, opcode: 0
0       odoh.cloudflare-dns.com.        IN      HTTPS   300     1 . alpn=h2 ipv4hint=104.16.248.249,104.16.249.249 ipv6hint=2606:4700::6810:f8f9,2606:4700::6810:f9f9 key32769=,�(  7_5:�)�*Ƀ�
�f�nD�0��'����P���T
$ sdig ::1 53 odoh.cloudflare-dns.com HTTPS recurse | tail -2 | hd
00000000  30 09 6f 64 6f 68 2e 63  6c 6f 75 64 66 6c 61 72  |0.odoh.cloudflar|
00000010  65 2d 64 6e 73 2e 63 6f  6d 2e 09 49 4e 09 48 54  |e-dns.com..IN.HT|
00000020  54 50 53 09 39 38 09 31  20 2e 20 61 6c 70 6e 3d  |TPS.98.1 . alpn=|
00000030  68 32 20 69 70 76 34 68  69 6e 74 3d 31 30 34 2e  |h2 ipv4hint=104.|
00000040  31 36 2e 32 34 38 2e 32  34 39 2c 31 30 34 2e 31  |16.248.249,104.1|
00000050  36 2e 32 34 39 2e 32 34  39 20 69 70 76 36 68 69  |6.249.249 ipv6hi|
00000060  6e 74 3d 32 36 30 36 3a  34 37 30 30 3a 3a 36 38  |nt=2606:4700::68|
00000070  31 30 3a 66 38 66 39 2c  32 36 30 36 3a 34 37 30  |10:f8f9,2606:470|
00000080  30 3a 3a 36 38 31 30 3a  66 39 66 39 20 6b 65 79  |0::6810:f9f9 key|
00000090  33 32 37 36 39 3d 00 2c  ff 03 00 28 00 20 00 01  |32769=.,...(. ..|
000000a0  00 01 00 20 37 5f 35 3a  e9 29 cc 2a c9 83 d8 0a  |... 7_5:.).*....|
000000b0  01 e2 66 ce 6e 44 c1 30  8b f2 27 94 80 86 d9 50  |..f.nD.0..'....P|
000000c0  f8 d7 e4 54 0a                                    |...T.|
000000c5
```

([`odoh.cloudflare-dns.com` announcement.](https://blog.cloudflare.com/oblivious-dns/))

### Other information
https://tools.ietf.org/html/draft-ietf-dnsop-svcb-https-02#section-2.1
https://tools.ietf.org/html/draft-ietf-dnsop-svcb-https-02#appendix-A

### PR DESCRIPTION
**Pull request by pieterlexis**
### Short description
There were multiple issues. With this commit, we *always* store the
bytes that are represented by the option. To do this, we needed to
properly parse RFC 1035 character-strings. This is now done with a
conversion of the ABNF from draft-ietf-dnsop-svcb-https-02 to ragel.

The resulting function could be used as a starting point for a better
TXT storage format.

Fixes #9829

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [x] documented the code
- [ ] added or modified regression test(s)
- [x] added or modified unit test(s)

### CODE REVIEW

File: pdns/dnslabeltext.rl [hunk 1]
```diff
@@ -169,6 +170,76 @@ DNSName::string_t segmentDNSNameRaw(const char* realinput, size_t inputlen)
...
 };
 
+// Reads an RFC 1035 character string from 'in', puts the resulting bytes in 'out'.
+// Returns the amount of bytes read from 'in'
+size_t parseRFC1035CharString(const std::string &in, std::string &val) {
+
+  val.clear();
+  val.reserve(in.size());
+  const char *p = in.c_str();
+  const char *pe = p + in.size() + 1;
```
[the remark this answers is not in the document]
Comment (pieterlexis) [reply pdns/dnslabeltext.rl:180]: I believe we would stop reading one character too soon

[the remark this answers is not in the document]
Comment (Habbie) [reply pdns/dnslabeltext.rl:180]: hmm, this smells :)

[the remark this answers is not in the document]
Comment (pieterlexis) [reply pdns/dnslabeltext.rl:180]: I'll make a bunch of tests for this function and adjust if needed

[the remark this answers is not in the document]
Comment (Habbie) [reply pdns/dnslabeltext.rl:180]: removing the +1 does not break the tests

[the remark this answers is not in the document]
Comment (Habbie) [reply pdns/dnslabeltext.rl:180]: putting in `-1` does break the tests :)

Comment (Habbie) [inline pdns/dnslabeltext.rl:180]:
```suggestion
  const char *pe = p + in.size();
```

[the remark this answers is not in the document]
Comment (Habbie) [reply pdns/dnslabeltext.rl]: I changed it, now going through CI

**Pull request merged**